minos/common/database/database.py

```diff
@@ -57,11 +52,11 @@ async def submit_query_and_iter(self, *args, **kwargs) -> AsyncIterator[tuple]:
         :param kwargs: Additional named arguments.
         :return: This method does not return anything.
         """
-        pool = await self.pool
-        with await pool.cursor() as cursor:
-            await cursor.execute(*args, **kwargs)
-            async for row in cursor:
-                yield row
+        async with self._pool.acquire() as connection:
+            with await connection.cursor() as cursor:
```

Comment:
Instead of a `connections`, we're acquiring a `pool`:
```suggestion
        async with self._pool.acquire() as pool:
            with await pool.cursor() as cursor:
```

Reply:
This suggestion is not correct